core/domain/event_services_test.py

```diff
@@ -382,14 +382,18 @@ def test_handle_event_method_is_called(self) -> None:
 class StatsEventsHandlerUnitTests(test_utils.GenericTestBase):
     """Tests related to the stats events handler."""
 
-    def test_stats_events_with_undefined_state_name_gets_logged(self) -> None:
+    @mock.patch('logging.error')
+    def test_stats_events_with_undefined_state_name_gets_logged(
+        self, mock_logging: mock.Mock
+    ) -> None:
         observed_log_messages = []
 
         def _mock_logging_function(msg: str, *args: str) -> None:
             """Mocks logging.error()."""
             observed_log_messages.append(msg % args)
 
-        logging_swap = self.swap(logging, 'error', _mock_logging_function)
+        mock_logging.side_effect = _mock_logging_function
+        logging_swap = mock_logging
```

Comment:
Why do you need this if we already have the patch applied?

Note that part of the point of this PR is to get rid of all the swapping.